src/pages/HomePage.tsx

```diff
@@ -0,0 +1,151 @@
+import React, {useEffect, useMemo, useState} from 'react';
+import {Button, StyleSheet, Text, View} from 'react-native';
+import {Address, Chain, createWalletClient, custom, formatEther} from 'viem';
+import {mainnet, sepolia} from 'viem/chains';
+import {useWalletConnectModal} from '@walletconnect/modal-react-native';
+import {publicClient} from '../clients/public';
+
+export const CHAINS = [mainnet, sepolia];
+
+type ConnectEventInfo = {
+  session: {
+    namespaces: {
+      eip155: {
+        chains: `eip155:${number}`[];
+      };
+    };
+  };
+};
+
+export default function HomePage() {
+  const {
+    open,
+    isConnected,
+    provider,
+    address: wcAddress,
+  } = useWalletConnectModal();
+  const address = wcAddress as Address | undefined;
+
+  const walletClient = useMemo(
+    () =>
+      createWalletClient({
+        chain: mainnet,
+        transport: custom({
+          async request({method, params}) {
+            return await provider?.request({method, params});
+          },
+        }),
+      }),
+    [provider],
+  );
+
+  const [blockNumber, setBlockNumber] = useState(0n);
+  const [gasPrice, setGasPrice] = useState(0n);
+  const [chain, setChain] = useState<Chain>(CHAINS[0]);
+  const [balance, setBalance] = useState(0n);
+  const [signature, setSignature] = useState<`0x${string}`>();
+
+  const onSignMessage = async () => {
+    if (address) {
+      const signature = await walletClient.signMessage({
+        account: address,
+        message: 'Sign this message to prove you are the owner of this wallet',
+      });
+
+      setSignature(signature);
+    }
+  };
+
+  useEffect(() => {
+    const getNetworkData = async () => {
+      const [blockNumber, gasPrice] = await Promise.all([
+        publicClient.getBlockNumber(),
+        publicClient.getGasPrice(),
+      ]);
+
+      setBlockNumber(blockNumber);
+      setGasPrice(gasPrice);
+    };
+
+    void getNetworkData();
+  }, []);
+
+  useEffect(() => {
+    const onChainChangedEvent = (chainId: string) => {
+      const chain =
+        CHAINS.find(chain => chain.id === Number(chainId)) ?? CHAINS[0];
+      setChain(chain);
+    };
+
+    const onConnectEvent = async (info: ConnectEventInfo) => {
+      const chainId = info.session.namespaces.eip155.chains[0].replace(
+        'eip155:',
+        '',
+      );
+
+      onChainChangedEvent(chainId);
+
+      if (address) {
+        const balance = await publicClient.getBalance({
+          address,
+        });
+        setBalance(balance);
+      }
+    };
+
+    // @ts-ignore
```

Comment:
@mdjastrzebski If you know a way to prevent this, let me know - I believe the provider types coming from WC are wrong